src/lib/cache/subCache.cpp

```diff
@@ -331,41 +337,38 @@ static bool attributeMatch(CachedSubscription* cSubP, const std::vector<std::str
 *
 * servicePathMatch -
 */
-static bool servicePathMatch(CachedSubscription* cSubP, char* servicePath)
+static bool servicePathMatch(CachedSubscription* cSubP, const std::string& servicePath)
 {
-  const char*  spath;
-
+  //
   // Special case. "/#" matches EVERYTHING
-  if ((strlen(servicePath) == 2) && (servicePath[0] == '/') && (servicePath[1] == '#') && (servicePath[2] == 0))
+  //
+  if (servicePath == "/#")
   {
     return true;
   }
 
   //
   // Empty service path?
   //
-  if ((strlen(servicePath) == 0) && (strlen(cSubP->servicePath) == 0))
+  if ((servicePath == "") && (cSubP->servicePath == ""))
```

Comment:
Maybe a more efficient way of checking string emptyness is to use the length() method, i.e.

```
if ((servicePath.length() == 0) && (cSubP->servicePath,lengt() == 0))
```

However, I'd suggest not chaning that in this PR, but move it to an external issue, as it could impact other places in the code.  In addition, if we agree on that, we should include it as rule in the coding style guide. 

(Probably the library implementation is clever enough so not sure if this could make any real difference... but probably worth the change anyway as it is easdy to do, i.e. just grep for `""` along the code).